configure.ac

```diff
@@ -182,15 +229,94 @@ if test -z "${MATH_LIBS}"; then
 	)
 fi
 
+if test "${with_math}" = "yes"; then
+	test -z "${MATH_LIBS}" && AC_MSG_ERROR([math required but not found])
+	AC_DEFINE([STORAGE_WITH_MATH], [1], [math settings])
+	OPTIONAL_MATH_CFLAGS="${MATH_CFLAGS}"
+	OPTIONAL_MATH_LIBS="${MATH_LIBS}"
+fi
+
+
+# -----------------------------------------------------------------------------
+# zlib
+
+if test "${with_zlib}" = "yes"; then
+	PKG_CHECK_MODULES(
+		[ZLIB],
+		[zlib],
+	)
+	test -z "${ZLIB_LIBS}" && AC_MSG_ERROR([zlib required but not found. Try installing 'zlib1g-dev' or 'zlib-devel'.])
+	AC_DEFINE([NETDATA_WITH_ZLIB], [1], [zlib settings])
+	OPTIONAL_ZLIB_CLFAGS="${ZLIB_CFLAGS}"
+	OPTIONAL_ZLIB_LIBS="${ZLIB_LIBS}"
+fi
+
+
+# -----------------------------------------------------------------------------
+# libuuid
+
 PKG_CHECK_MODULES(
 	[UUID],
 	[uuid],
+	[have_uuid=yes],
+	[have_uuid=no]
 )
-test -z "${UUID_LIBS}" && AC_MSG_ERROR([libuuid required but not found. Try installing 'uuid-dev' or 'libuuid-devel'.])
-AC_DEFINE([NETDATA_WITH_UUID], [1], [uuid settings])
+test "${have_uuid}" = "no" && AC_MSG_ERROR([libuuid required but not found. Try installing 'uuid-dev' or 'libuuid-devel'.])
+AC_DEFINE([NETDATA_WITH_UUID], [1], [uuid usability])
 OPTIONAL_UUID_CLFAGS="${UUID_CFLAGS}"
 OPTIONAL_UUID_LIBS="${UUID_LIBS}"
 
+
+# -----------------------------------------------------------------------------
+# libcap
+
+if test "${with_cap}" != "no"; then
+	PKG_CHECK_MODULES(
+		[CAP],
+		[libcap],
+		[have_cap=yes],
+		[have_cap=no]
+	)
+fi
+test "${have_cap}" = "no" -a "${with_cap}" = "yes" && AC_MSG_ERROR([libcap required but not found.])
+if test "${have_cap}" = "yes"; then
+	AC_DEFINE([HAVE_CAPABILITY], [1], [libcap usability])
+	OPTIONAL_CAP_CLFAGS="${CAP_CFLAGS}"
+	OPTIONAL_CAP_LIBS="${CAP_LIBS}"
+fi
+AM_CONDITIONAL([ENABLE_CAPABILITY], [test "${have_cap}" = "yes"])
```

Comment:
good!
but btw, in this case you can PKG_CHECK_MODULES without the conditional and it still work the same.
using PKG_CHECK_MODULES is nice as it does not fail like checking lib/header and less conditionals the better.
```
PKG_CHECK_MODULES(
    [CAP], # why not LIBCAP all over to be consistent?
    [libcap],
    [have_cap=yes],
    [have_cap=no]
)
test "${have_cap}" = "no" -a "${with_cap}" = "yes" && AC_MSG_ERROR([libcap required but not found.])
if test "${enable_pcap}" != "no"; then
    enable_cap="yes"
    AC_DEFINE([HAVE_CAPABILITY], [1], [libcap usability])
    OPTIONAL_CAP_CLFAGS="${CAP_CFLAGS}"
    OPTIONAL_CAP_LIBS="${CAP_LIBS}"
fi
AM_CONDITIONAL([ENABLE_CAPABILITY], [test "${enable_cap}" = "yes"])
```